### PR TITLE
Task 9

### PR DESCRIPTION
**Pull request by trigolos**
Task: https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task9-bff/task.md

Products:
BFF URL - http://trigolos-bff-api-develop.eu-west-1.elasticbeanstalk.com/products
Original URL - https://tsz5wbnhm9.execute-api.eu-west-1.amazonaws.com/dev/products

Cart:
BFF URL - http://trigolos-bff-api-develop.eu-west-1.elasticbeanstalk.com/cart
Original URL - http://trigolos-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart

How to create product:
Send POST request with [Postman ](https://www.postman.com/downloads/) to http://trigolos-bff-api-develop.eu-west-1.elasticbeanstalk.com/products (or https://tsz5wbnhm9.execute-api.eu-west-1.amazonaws.com/dev/products)
Body example: 
`{
    "title": "Title",
    "description": "Description",
    "author": "Author",
    "price": 123,
    "count": 1000
}`
image

Evaluation criteria:
- [x] - A working and correct express application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should works
- [x] - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or CART. The response from the bff-service should be the same as if product-service or CART services were called directly.

Additional (optional) tasks:
- [x] - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
- [ ]  Use NestJS to create bff-service instead of express

### CODE REVIEW

File: bff-service/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            method: req.method as Method,
+            url: `${recipientUrl}${req.originalUrl}`,
+            ...(Object.keys(req.body || {}).length > 0 && { data: req.body }),
+        };
+
+        console.log('axiosConfig: ', axiosConfig);
+
+        try {
+            const result = await request(axiosConfig);
+            res.json(result);
```
Comment (IvanPresmytsky) [inline bff-service/src/index.ts:35]: You missed sending status here

**Review by IvanPresmytsky — APPROVED** [review]
Good job!

File: bff-service/src/index.ts (hunk 1 above)
Comment (trigolos) [reply bff-service/src/index.ts:35]: As I understand, here we need a status code of 200, that is code by default (https://nodejs.org/api/http.html#http_response_statuscode)

**Review by skyRoma — APPROVED** [review]

File: bff-service/src/index.ts (hunk 1 above)
Comment (IvanPresmytsky) [reply bff-service/src/index.ts:35]: How about POST request? :)

Comment (trigolos) [reply bff-service/src/index.ts:35]: What's wrong with POST request? Product creation works correctly

Comment (IvanPresmytsky) [reply bff-service/src/index.ts:35]: I mean, you will need a different status for successfull POST request.

Comment (trigolos) [reply bff-service/src/index.ts:35]: Yeah, you are right, thanks. Fogot about 201 status code for creating

**Pull request merged**